server.cpp

```diff
@@ -38,6 +38,19 @@ std::string make_daytime_string()
     return buffer;
 }
 
+std::string int_to_hex(int dec)
+{
+    std::string digits[16] = {"0","1","2","3","4","5","6","7","8","9","A","B","C","D","E","F"};
+    std::string hex;
+    do
+    {
+        hex.insert (0, digits[dec % 16]);
+        dec /= 16;
+    }
+    while (dec != 0);
+    return "0x" + hex;
```

Comment:
According to https://tools.ietf.org/html/rfc2616#section-3.6.1 chunk size is defined as a sequence of HEX digits, so it cannot have '0x' prefix.